ts/features/idpay/onboarding/screens/InitiativeDetailsScreen.tsx

```diff
@@ -97,89 +110,131 @@ const BeforeContinueBody = (props: BeforeContinueBodyProps) => {
+  }: NativeSyntheticEvent<NativeScrollEvent>) => {
+    const { layoutMeasurement, contentOffset, contentSize } = nativeEvent;
+    const paddingToBottom = 20;
+    if (
+      layoutMeasurement.height + contentOffset.y >=
+      contentSize.height - paddingToBottom
+    ) {
+      setHasScrolled(true);
+    }
+  };
+  const scrollViewRef = React.useRef<ScrollView>(null);
+  const handleGoBackPress = () => {
+    machine.send({ type: "QUIT_ONBOARDING" });
+  };
+  const handleContinuePress = () =>
+    isContinueButtonDisabled
+      ? scrollViewRef.current?.scrollToEnd()
+      : machine.send({ type: "ACCEPT_TOS" });
 
-  React.useEffect(() => {
-    send({
-      type: "SELECT_INITIATIVE",
-      serviceId
-    });
-  }, [send, serviceId]);
-
-  // TODO show initaitveID for testing purposes
-  const content = pipe(
-    O.fromNullable(state.context.initiative),
-    O.map(initiative => `Initiative ID: ${initiative.initiativeId}`),
+  const service = pipe(
+    pot.toOption(
+      useIOSelector(serviceByIdSelector(serviceId as ServiceId)) || pot.none
+    ),
     O.toUndefined
   );
 
+  const setMarkdownIsLoaded = () => (isMarkdownLoadedRef.current = true);
+
+  const screenContent = () => (
+    <SafeAreaView style={IOStyles.flex}>
+      <ScrollView
+        onLayout={handleScrollViewLayout}
+        onContentSizeChange={handleScrollViewContentSizeChange}
+        onScroll={handleScrollViewOnScroll}
+        scrollEventThrottle={400}
+        ref={scrollViewRef}
+        style={IOStyles.flex}
+      >
+        <View style={IOStyles.horizontalContentPadding}>
+          {service !== undefined && (
+            <InitiativeOrganizationHeader {...toUIService(service)} />
+          )}
+          <VSpacer size={16} />
+          {description !== undefined && (
+            <Markdown onLoadEnd={setMarkdownIsLoaded}>{description}</Markdown>
+          )}
+          <VSpacer size={16} />
+          <ItemSeparatorComponent noPadded={true} />
+          <VSpacer size={16} />
+          <BeforeContinueBody
+            tosUrl={service?.service_metadata?.tos_url}
+            privacyUrl={service?.service_metadata?.privacy_url}
+          />
+        </View>
+        <VSpacer size={16} />
+      </ScrollView>
+      <FooterWithButtons
+        type={"TwoButtonsInlineThird"}
+        leftButton={{
+          bordered: true,
+          title: I18n.t("global.buttons.cancel"),
+          onPress: handleGoBackPress,
+          testID: "IDPayOnboardingCancel"
+        }}
+        rightButton={{
+          title: I18n.t("global.buttons.continue"),
+          onPress: handleContinuePress,
+          testID: "IDPayOnboardingContinue",
+          style: isContinueButtonDisabled
+            ? {
+                flex: 2,
+                backgroundColor: IOColors.grey
+              }
+            : { flex: 2 },
+          // the style prop overrides the default style of the button,
+          // so we need to add the flex: 2 property for consistency
```

Comment:
We should not let the user interact with the button if it is (or it seems) disabled.

Reply:
so you are confirming we should scrap the "tap to scroll" functionality?

Reply:
Yes, I'd suggest drafting this PR as we don't yet know how to fix this.